clients/python-wrapper/lakefs/config.py

```diff
@@ -60,35 +107,162 @@ def __init__(self, verify_ssl: Optional[bool] = None, proxy: Optional[str] = Non
 
         if kwargs:
             return
+        self._load_from_config_file()
+        self._load_from_environment()
+        # Check for IAM provider if no session token and no credentials
+        if self.access_token is None and self.username is None and self.password is None:
+            self._iam_provider = get_iam_provider_from_env_or_file(self._config_data)
 
-        found = False
+        if not self._has_valid_authentication():
+            raise NoAuthenticationFound
 
-        # Get credentials from lakectl
+    def _load_from_config_file(self):
+        """Load configuration from .lakectl.yaml file if it exists"""
         try:
             with open(_LAKECTL_YAML_PATH, encoding="utf-8") as fd:
                 data = yaml.load(fd, Loader=yaml.Loader)
-                self.server = (ClientConfig.Server(**data["server"])
-                               if "server" in data
-                               else ClientConfig.Server(endpoint_url=""))
-                self.credentials = (ClientConfig.Credentials(**data["credentials"])
-                                   if "credentials" in data
-                                   else ClientConfig.Credentials(access_key_id="", secret_access_key=""))
-                found = True
-        except FileNotFoundError:  # File not found, fallback to env variables
-            self.server = ClientConfig.Server(endpoint_url="")
-            self.credentials = ClientConfig.Credentials(access_key_id="", secret_access_key="")
-
-        endpoint_env = os.getenv(_LAKECTL_ENDPOINT_ENV)
+                if "server" in data:
+                    self.server = ClientConfig.Server(**data["server"])
+                if "credentials" in data:
+                    try:
+                        self.credentials = ClientConfig.Credentials(**data["credentials"])
+                        self.username = self.credentials.access_key_id
+                        self.password = self.credentials.secret_access_key
+                    except TypeError:
+                        pass
+                self._config_data = data
```

Comment:
You have code paths in your init where the _config_data is not initialized explicitly
I don't see it being used outside the scope of the init func so I'm wondering why is it a data member of the instance - use a local variable instead